### PR TITLE
Adds Fail rescue strategy

### PR DESCRIPTION
**Pull request by adamruzicka**
- [x] tests

### CODE REVIEW

**Comment by adamruzicka** [comment]
Test failures seem to be unrelated

File: lib/dynflow/action/rescue.rb [hunk 1]
```diff
@@ -44,14 +44,20 @@ def rescue_strategy_for_planned_action(action)
...
-            suggested_strategies.all? { |suggested_strategy| suggested_strategy.strategy == Skip }
+      if suggested_strategies.empty?
         return Skip
       else
-        return Pause
+        # TODO: Find the safest rescue strategy among the suggested ones
+        if suggested_strategies.all? { |suggested_strategy| suggested_strategy.strategy == Skip }
+          return Skip
+        elsif suggested_strategies.any? { |suggested_strategy| suggested_strategy.strategy == Fail }
+          return Fail
```
Comment (iNecas) [inline lib/dynflow/action/rescue.rb]: I'm not sure if one sub-action somewhere at the bottom of the plan tree should be able to set the whole task as failed. Notice that in Skip, all the sub-plans need to agree on the skip before proceeding, Also, the fail strategy is influencing also the steps that were not started yet. I think, for this strategy, should win only if all the pending steps are planned by the action that suggests the Fail strategy.

Comment (adamruzicka) [inline lib/dynflow/action/rescue.rb]: Changed to the same behavior as used for `Skip`

**Comment by adamruzicka** [comment]
We're green

**Comment by iNecas** [comment]
Looks good, please squash so that we can merge it

**Comment by adamruzicka** [comment]
@inecas squashed and tests are green

**Comment by iNecas** [comment]
Thanks @adamruzicka!

**Pull request merged**